lib/util/player_configuration.js

```diff
@@ -247,6 +247,7 @@ shaka.util.PlayerConfiguration = class {
         slowHalfLife: 5,
       },
       restrictToElementSize: false,
+      restrictToScreenSize: false,
```

Comment:
What is the use case for this feature? It seems like `restrictToElementSize` covers this functionality. If `restrictToScreenSize` is `true` and `restrictToElementSize` is `false`, wouldn't that lead to a scenario where a 320x180 player could be downloading and playing 4k?

Reply:
The main difference is that one is not dynamic and the other is (ResizeObserver). If you have a small 240p video and the screen is 1080p the behavior with the new thing is that it can reach 1080p in the small video and when it is set to full screen there is no difference.
The ABR regarding restrictToElementSize is not fully dynamic in that if you change the size the video doesn't start rendering at that size but instead adds to the already existing buffer.